Library/Homebrew/formula_cellar_checks.rb

```diff
@@ -325,6 +325,7 @@ def check_binary_arches(formula)
       farch = file.arch
       mismatches[file] = farch unless farch == Hardware::CPU.arch
     end
+    mismatches.reject! { |f| f.first.to_s.start_with? formula.share }
```

Comment:
Let's also add a comment here that we ignore `share` because some formulae (e.g. `lima`) intentionally install non-native binaries into `share`.

That said, `share` is typically used for [architecture-independent data](https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard#Directory_structure) so I'm now wondering if this is indeed the right approach here...

Reply:
Updated and tested on Ubuntu.